cmd/readPipelineEnv.go

```diff
@@ -58,10 +61,10 @@ func runReadPipelineEnv(config *artifactPrepareVersionOptions) error {
 	}
 
 	// try to encrypt
-	if config.Password != "" && orchestrator.DetectOrchestrator() == orchestrator.GitHubActions {
+	if encryptedCPE && stepConfigPassword != "" {
```

Comment:
If the password is not there, the CPE will without notice not be encrypted despite the flag being set to true. This can result in unexpected exposure of secrets. I think it's better to throw an error in this case.